index_module.inc.php

```diff
@@ -69,33 +69,39 @@
             }
 
             //// Load Mod-Config
+            $modParameter = $request->query->get('mod');
+            $actionParameter = $request->query->get('action');
             // 1) Search $_GET['action'] in DB (field "action")
-            $menu = $db->qry_first("SELECT file, requirement FROM %prefix%menu WHERE (module = %string%) and (action = %string%)", $_GET['mod'], $_GET['action']);
-            if ($menu['file'] != '') {
+            $menu = $db->qry_first("SELECT file, requirement FROM %prefix%menu WHERE (module = %string%) and (action = %string%)", $modParameter, $actionParameter);
+            if ($menu && $menu['file'] != '') {
                 if ($authentication->authorized($menu['requirement'])) {
-                    include_once("modules/{$_GET['mod']}/{$menu['file']}.php");
+                    // TODO Fix security issue
+                    include_once("modules/{$modParameter}/{$menu['file']}.php");
```

Comment:
That is actually a fair point, In how far are the values from httpfoundation sanitized? not saying that  ``addslashes()`` provided the most secure environment here, but it was at least ...something.

Did not see any direct hint in the symphony documentation that the user input is sanitized for general use, so are we OK with just passing this along?
Or is this also still ok due to the original array processing in index.php?

Reply:
The issue here with adding a `$_GET` parameter into a path is a possible [path / directory traversal](https://owasp.org/www-community/attacks/Path_Traversal) (e.g. `../../../etc/passwd`).

`addslashes()` has nothing to do with this attack because it only escapes

* single quote (')
* double quote (")
* backslash (\)
* NUL (the NUL byte)

The Symfony HTTPFoundation component is not having any security protection built in for this. Mainly because the component doesn't know in which context the variable/output is used. Even with the usage of the component, we might still be vulnerable to path/directory traversal. 
One of the main benefits of the component is (next to unified access), that we don't need to have multiple checks if the parameter exists and is filled, etc. to avoid access to arrays that don't contain the requested key.
See https://github.com/symfony/http-foundation/blob/718a97ed430d34e5c568ea2c44eab708c6efbefb/InputBag.php#L28

Personally, I am not OK with having user-controlled input in such an include statement. I did not check if this is a vuln that can be attacked. That's why I added a TODO comment there + to not blow up this PR any further.
I suggest to investigate this in an additional PR.